src/array.cr

```diff
@@ -1808,6 +1811,21 @@ class Array(T)
     insert 0, obj
   end
 
+  # Prepend multiple values. The same as `unshift`, but takes an arbitrary number
+  # of values to add to the array. Returns `self`.
+  def unshift(*values : T)
+    new_size = @size + values.size
+    resize_to_capacity(Math.pw2ceil(new_size)) if @size > @capacity
+    move_value = values.size
+    @buffer.move_from(@buffer - move_value, @size + move_value)
```

Comment:
Actually, that's wrong too. The correct one is:

```crystal
@buffer.move_to(@buffer + move_value, @size)
```

We want to move everything we have in the array (`@size`) `move_value` positions to the left (`@buffer + move_value`), starting from the beginning of the array (`@buffer`)